codegen/src/SchemaProvider.scala

```diff
@@ -53,11 +39,44 @@ class SchemaProvider(schemaCacheDirPath: os.Path) {
       moduleFormat = pulumiPackage.meta.moduleFormat.r
     )
   }
+}
+
+class TestSchemaProvider(schemaPath: os.Path) extends SchemaProvider {
```

Comment:
I think we shouldn't try to special case on making a schema provider return schemas just from one file - we can still rely on the logic which accepts a path to a directory where we can have json schemas for different providers in different versions - we would just need to keep this directory structure in test resources. For tests we might just want to prevent the provider from trying to download schemas if not found locally.
I suppose this approach would make test cases like `external-enum` work, because they contain inter-schema references to types

Reply:
I've made an issue to remember to follow up with the rework, thank you.
https://github.com/VirtusLab/besom/issues/257